### DISCUSSION TITLE
[Question] How to create a sortable Table in a Tiddler?

### DISCUSSION THREAD

**Discussion by Parlazo** [TiddlyWiki/TiddlyWiki5#5680] [Q&A]
Is there a way to create a sortable table in a Tiddler?

i would like to use a html or any format table generated with https://www.tablesgenerator.com/html_tables#
below is a button which allows to make table sortable, this will insert a javascript
if i copy the code in a tiddler it will strip the css styles and the javascript, if i insert the code into a html tiddler it will keep the style but still strip the javascript

the best would be to keep the style and with some surrounding magic widget the whole table would be sortable and styled in a unboxed tw5 tiddler

any suggestion how this could work?

Thanks

**Comment by twMat** [comment]
>Is there a way to create a sortable table in a Tiddler?

@Parlazo 
To put it in a simplified way: It is easy to create a table where *one* column is sortable, e.g a column showing the tiddler titles.
Would this be enough for you?

**Reply by Parlazo** [reply]
one column? ... any one? or every one?

i have some columns and i would like to make the table sortable by all colums but one at a time... so not a sort like one column and additional the other column ... but one or an other
this would be perfect

in example

```
col1 col2 col3
a    3    beta
b    2    alpha
c    1    gamma
```

and now sort by abc or 123 or alpha beta gamma ... but not all together... just one or the other column while the column header will stay on top.
and i also would like to use a table with typed data not necessary a data tiddler but if it could contain tiddler titles i think it would be fine too.

:smiley:

**Reply by twMat** [reply]
>one column? ... any one? or every one?

Exactly, that's why I asked. Doing it with one specific column (e.g always column 3) is simple. Not simple what you request. But it has been discussed and I believe there are solutions that you can find via the [google forum](https://groups.google.com/g/tiddlywiki/search?q=sortable%20table).
